node/pkg/watchers/solana/shim.go

```diff
@@ -269,17 +282,25 @@ func (s *SolanaWatcher) shimProcessRest(
 	var verifiedCoreEvent bool
 	var messageEvent *ShimMessageEventData
 	var err error
-	coreEventFound := false
 	for idx := startIdx; idx < len(innerInstructions); idx++ {
 		inst := innerInstructions[idx]
 		if inst.ProgramIDIndex == whProgramIndex {
-			if verifiedCoreEvent, err = shimVerifyCoreMessage(inst.Data); err != nil {
+			if verifiedCoreEvent {
+				return fmt.Errorf("detected multiple inner core instructions when there should not be at instruction %d, %d", outerIdx, idx)
```

Comment:
Right now I believe this check is too general: this returns an error if the inner instruction's program index matches the Wormhole Program's, regardless of the contents. I think instead what should be flagged here is the case where another "verified core event" occurs.

So the more targeted check would be something like:

```
// Ensure that only one verified core event appears in each set of inner instructions
if foundIt && verifiedCoreEvent {
 // return error 
 }
```

In practice this might be fine, but maybe there's a valid case (now or down the line) where the Wormhole program needs to be interacted with twice in the same inner instruction set.